ipaserver/install/plugins/update_ra_cert_store.py

```diff
@@ -0,0 +1,115 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import binascii
+import os
+import pwd
+
+from ipalib import Registry
+from ipalib import Updater
+from ipalib.install import certmonger
+from ipaplatform.constants import constants
+from ipaplatform.paths import paths
+from ipaplatform.tasks import tasks
+from ipapython import certdb
+
+register = Registry()
+
+
+@register()
+class update_ra_cert_store(Updater):
+    """
+    Moves the cert store from /etc/httpd/alias to /var/lib/ipa/radb
+    """
+
+    def execute(self, **options):
+        olddb = certdb.NSSDatabase(nssdir=paths.HTTPD_ALIAS_DIR)
+        if not olddb.has_nickname('ipaCert'):
+            # Nothign to do
+            return False, []
+
+        newdb = certdb.NSSDatabase(nssdir=paths.IPA_RADB_DIR)
+        if os.path.exists(paths.IPA_RADB_DIR):
+            if newdb.has_nickname:
+                self.log.warning(
+                    "An 'ipaCert' nickname exists in both the old {} and the "
+                    "new {} NSS Databases!".format(paths.HTTPD_ALIAS_DIR,
+                                                   paths.IPA_RADB_DIR))
+                return False, []
+        else:
+            # Get the owning user ids
+            pent = pwd.getpwnam(constants.IPAAPI_USER)
+
+            # create the initial directory
+            os.makedirs(paths.IPA_RADB_DIR, 0o751)
+            os.chown(paths.IPA_RADB_DIR, pent.pw_uid, pent.pw_gid)
+            tasks.restore_context(paths.IPA_RADB_DIR)
+
+            # Create the password file for this db
+            hex_str = binascii.hexlify(os.urandom(10))
+            f = os.open(paths.IPA_RADB_PWDFILE_TXT, os.O_CREAT | os.O_RDWR)
+            os.write(f, hex_str)
+            os.close(f)
+
+            # And create the DB
+            newdb.create_db(paths.IPA_RADB_PWDFILE_TXT)
+
+        # Import cert chain (ignore errors, as certs may already be imported)
+        certlist = olddb.list_certs()
+        certflags = {}
+        for name, flags in certlist:
+            certflags[name] = flags
+        for name in olddb.get_trust_chain('ipaCert'):
+            if name == 'ipaCert':
+                continue
+            try:
+                cert = olddb.get_cert(name, pem=True)
+                newdb.add_cert(cert, name, certflags[name], pem=True)
+            except Exception as e:  # pylint disable=broad-except
+                self.log.warning("Failed to import '{}' from trust "
+                                 "chain: {}".format(name, str(e)))
```

Comment:
My understanding is that this code is built this way in order to preserve the original trust flags. I copied it from another part of the code where this was being done. I am wary of changing it because it seem to have a well meaning purpose.